elftools/dwarf/dwarf_util.py

```diff
@@ -60,3 +60,18 @@ def _iter_CUs_in_section(stream, structs, parser):
         yield header
         offset = header.offset_after_length + header.unit_length   
 
+def _file_crc32(file):
+    """ Provided a readable binary file stream, reads the whole file
+        and computes the CRC32 checksum of its contents,
+        with the initial value of 0.
+
+        Assumes the file pointer is at the top, and rewinds to the top
```

Comment:
This makes me realize that the whole stream setting here is a bit magical and may not be generally useful. How about using `preserve_stream_pos` here, or just at the call site if the call site wants to preserve the position?

Reply:
The one and only usage point calls this before an `ELFFile` is constructed over the stream - which also assumes a stream at the beginning. Who knows what might other usage points (if any, if ever) require. I'll move the `seek()` to the call site.